README.md

```diff
@@ -3,20 +3,35 @@
 [![Build Status](https://travis-ci.org/dart-lang/code_builder.svg)](https://travis-ci.org/dart-lang/code_builder)
 [![Coverage Status](https://coveralls.io/repos/dart-lang/code_builder/badge.svg)](https://coveralls.io/r/dart-lang/code_builder)
 
-Code builder is a fluent Dart API for generating valid Dart source code.
+`code_builder` is a fluent Dart API for generating valid Dart source code.
 
-Generally speaking, code generation usually is done through a series of
-string concatenation which results in messy and sometimes invalid code
-that is not easily readable.
+Code generation was traditionally done through a series of 
+package-specific string concatenations which usually results in messy
+and sometimes invalid Dart code that is not easily readable and is very
+difficult to refactor.
 
-Code builder uses the [analyzer](analyzer) package to create real Dart
+`code_builder` uses the [analyzer](analyzer) package to create real Dart
 language ASTs, which, while not guaranteed to be correct, always follows
 the analyzer's own understood format.
 
 [analyzer]: https://pub.dartlang.org/packages/analyzer
 
-Code builder also adds a more narrow and user-friendly API. For example
-creating a class with a method is an easy affair:
+## Experimental
+
+While `code_builder` is considered *stable*, the APIs are subject to
+frequent breaking change - a number of Dart language features are not
+yet implemented that make it unsuitable for all forms of code
+generation. 
+
+**Contributions are [welcome][welcome]!**
```

Comment:
`[foo][bar]` means that `[bar]` is defined lower in the markdown document, i.e:

`[bar]: http://some-long-link-that-is-awkward-inline.com`